apps/builder/app/builder/features/settings-panel/props-section/animation/animation-panel-content.tsx

```diff
@@ -0,0 +1,409 @@
+          return;
+        }
+
+        setIntermediateValue({
+          type: "invalid",
+          value: toValue(event.value),
+        });
+      }}
+      onAbort={() => {
+        /* @todo: allow to change some ephemeral property to see the result in action */
+      }}
+      onReset={() => {
+        setIntermediateValue(undefined);
+      }}
+    />
+  );
+};
+
+const EasingInput = ({
+  id,
+  value,
+  onChange,
+}: {
+  id: string;
+  value: string | undefined;
+  onChange: (value: string | undefined) => void;
+}) => {
+  const [intermediateValue, setIntermediateValue] = useState<
+    StyleValue | IntermediateStyleValue
+  >();
+
+  const property = "animationTimingFunction";
+
+  return (
+    <CssValueInput
+      id={id}
+      styleSource="default"
+      value={
+        value === undefined
+          ? { type: "keyword", value: "ease" }
+          : { type: "unparsed", value }
+      }
+      getOptions={() => [
+        ...styleConfigByName(property).items.map((item) => ({
+          type: "keyword" as const,
+          value: item.name,
+        })),
+      ]}
+      property={property}
+      intermediateValue={intermediateValue}
+      onChange={(styleValue) => {
+        setIntermediateValue(styleValue);
+        /* @todo: allow to change some ephemeral property to see the result in action */
+      }}
+      onHighlight={() => {
+        /* Nothing to Highlight */
+      }}
+      onChangeComplete={(event) => {
+        onChange(toValue(event.value));
+        setIntermediateValue(undefined);
+      }}
+      onAbort={() => {
+        /* @todo: allow to change some ephemeral property to see the result in action */
+      }}
+      onReset={() => {
+        setIntermediateValue(undefined);
+      }}
+    />
+  );
+};
+
+export const AnimationPanelContent = ({ onChange, value, type }: Props) => {
+  const fieldIds = useIds([
+    "rangeStartName",
+    "rangeStartValue",
+    "rangeEndName",
+    "rangeEndValue",
+    "fill",
+    "easing",
+  ] as const);
+
+  const timelineRangeDescriptions =
+    type === "scroll" ? scrollTimelineRangeName : viewTimelineRangeName;
+
+  const timelineRangeNames = Object.keys(timelineRangeDescriptions);
+
+  const animationSchema =
+    type === "scroll" ? scrollAnimationSchema : viewAnimationSchema;
+
+  const handleChange = (rawValue: unknown) => {
+    const parsedValue = animationSchema.safeParse(rawValue);
+    if (parsedValue.success) {
+      onChange(parsedValue.data);
+      return;
+    }
+
+    toast.error("Animation schema is incompatible, try fix");
+  };
+
+  return (
+    <Grid gap="2" css={{ padding: theme.panel.padding }}>
+      <Grid gap={1} align={"center"} css={{ gridTemplateColumns: "1fr 1fr" }}>
+        <Label htmlFor={fieldIds.fill}>Fill Mode</Label>
+        <Label htmlFor={fieldIds.easing}>Easing</Label>
+        <Select
+          id={fieldIds.fill}
+          options={fillModeNames}
+          getLabel={(fillModeName: string) => toPascalCase(fillModeName)}
```

Comment:
Maybe titleCase to have it space separated?

Reply:
done